src/Core/Form/IdentifiableObject/CommandBuilder/Product/SpecificPricePriorityCommandsBuilder.php

```diff
@@ -0,0 +1,59 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Form\IdentifiableObject\CommandBuilder\Product;
+
+use PrestaShop\PrestaShop\Core\Domain\Product\SpecificPrice\Command\RemoveSpecificPricePriorityForProductCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\SpecificPrice\Command\SetSpecificPricePriorityForProductCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductId;
+
+class SpecificPricePriorityCommandsBuilder implements ProductCommandsBuilderInterface
+{
+    public function buildCommands(ProductId $productId, array $formData): array
+    {
+        if (!isset($formData['pricing']['priority_management'])) {
+            return [];
+        }
+
+        $prioritiesData = $formData['pricing']['priority_management'];
+
+        if (empty($prioritiesData['use_custom_priority'])) {
+            return [new RemoveSpecificPricePriorityForProductCommand($productId->getValue())];
+        }
+
+        if (!empty($prioritiesData['priorities'])) {
```

Comment:
same remark as ['use_custom_priority'] for ['priorities']

Reply:
if we custom_priorities is set, then it comes here and sets the actual custom priorities. (!empty check is just some additional safecheck, it probably would only happen if something goes wrong in frontend)